### PR TITLE
Added Linear Regression + MANCOVA

### PR DESCRIPTION
**Pull request by raviselker**

### CODE REVIEW

File: R/linreg.b.R [hunk 1]
```diff
@@ -5,11 +5,534 @@ linRegClass <- R6::R6Class(
     "linRegClass",
     inherit = linRegBase,
     private = list(
+        #### Member variables ----
```
Comment (jonathon-love) [inline R/linreg.b.R:6]: you can remove the `# This file is a generated template, your changes will not be overwritten`

File: R/mancova.b.R [hunk 2]
```diff
@@ -5,11 +5,367 @@ mancovaClass <- R6::R6Class(
     "mancovaClass",
```
Comment (jonathon-love) [inline R/mancova.b.R:3]: you can remove the: `# This file is a generated template, your changes will not be overwritten`

File: jamovi/linreg.a.yaml [hunk 3]
```diff
@@ -25,16 +56,167 @@ options:
...
       type: Array
       default: [[]]
       template:
           type: Variables
       description:
           R: >
-            a list of lists, where each list describes the `label` (as a
-            string) and the `levels` (as vector of strings) of a particular
-            repeated measures factor
+            the blocks that go into the model
```
Comment (jonathon-love) [inline jamovi/linreg.a.yaml]: this could be slightly more descriptive (we always explain the type for example)

File: jamovi/linreg.a.yaml [hunk 4]
```diff
@@ -25,16 +56,167 @@ options:
...
+
+    - name: fitMeasures
+      title: Fit Measures
+      type: NMXList
+      options:
+          - name: r
+            title: R
+          - name: rSqr
```
Comment (jonathon-love) [inline jamovi/linreg.a.yaml]: you don't want to use `R2`?

File: jamovi/linreg.r.yaml [hunk 5]
```diff
@@ -4,10 +4,346 @@ title: Linear Regression
 jrs: '1.0'
```
Comment (jonathon-love) [inline jamovi/linreg.r.yaml]: so this files permissions have changed from 644 to 755. i'm pretty sure we don't want that. i'm a bit puzzled as to how that could happen.

File: jamovi/linreg.u.yaml [hunk 6]
```diff
@@ -1,13 +1,20 @@
 title: Linear Regression
```
Comment (jonathon-love) [inline jamovi/linreg.u.yaml:1]: funny perms here too.

File: jamovi/mancova.a.yaml [hunk 7]
```diff
@@ -4,16 +4,139 @@ title:   MANCOVA
 menuGroup: ANOVA
 
 version: '1.0.0'
-jas: '1.1'
+jas: '1.0'
```
Comment (jonathon-love) [inline jamovi/mancova.a.yaml]: `jas` and `jrs` should all be 1.1 now

File: jamovi/mancova.a.yaml [hunk 8]
```diff
@@ -4,16 +4,139 @@ title:   MANCOVA
...
+
+    - name: fixedFactors
+      title: Fixed factors
+      type: Variables
+      suggested:
+        - nominal
+        - ordinal
+      default:
+      description: >
+        The fixed factors
```
Comment (jonathon-love) [inline jamovi/mancova.a.yaml]: these guys should be more like our other descriptions

File: jamovi/mancova.a.yaml [hunk 9]
```diff
@@ -4,16 +4,139 @@ title:   MANCOVA
...
       type: Variables
+      suggested:
+        - continuous
+      permitted:
+        - continuous
+        - nominal
+        - ordinal
+      default:
+      description: >
+        The covariates
```
Comment (jonathon-love) [inline jamovi/mancova.a.yaml]: this guy too

**Review by jonathon-love — COMMENTED** [review]
looking good!

File: jamovi/linreg.r.yaml (hunk 5 above)
Comment (jonathon-love) [reply jamovi/linreg.r.yaml]: `chmod 644 filename`

File: R/linreg.b.R (hunk 1 above)
Comment (raviselker) [reply R/linreg.b.R:6]: Fixed

File: R/mancova.b.R (hunk 2 above)
Comment (raviselker) [reply R/mancova.b.R:3]: Fixed

File: jamovi/linreg.u.yaml (hunk 6 above)
Comment (raviselker) [reply jamovi/linreg.u.yaml:1]: No idea why this happened, but should be fixed with my latest PR.

**Comment by jonathon-love** [comment]
merged

**Pull request closed without merging**